logins-api/android/library/src/main/java/org/mozilla/sync15/logins/SyncResult.kt

```diff
@@ -123,11 +123,17 @@ class SyncResult<T> {
 
             override fun onException(exception: Exception) {
```

Comment:
Hmm, this `FooResult` abstraction is starting to need its own test suite!  @thomcc, can you own sharing this result type across `fxa-client` and `logins` (and, soon, `places`)?  With the layer-cake that we're implementing we will be able to ship a little shared helper library as part of `fxa-client` in `a-s` (and `logins`), and access it from `a-c`.  That is, we can move the result type out of `a-c` into this repo, hopefully solving some of our duplication issues.  And presumably we can also share an underlying Rust type.

This probably just needs a ticket filed and some discussion for now.

Reply:
I think the path forward is to expose a synchronous API from here and consume it as async in android-components. I'll file a bug though.